src/systems/dynamic_detachable_joint/DynamicDetachableJoint.cc

```diff
@@ -0,0 +1,386 @@
+        this->detachableJointEntity = _ecm.CreateEntity();
+
+        // creating the joint
+        _ecm.CreateComponent(
+            this->detachableJointEntity,
+            components::DetachableJoint({this->parentLinkEntity,
+                                         this->childLinkEntity, "fixed"}));
+        this->attachRequested = false;
+        this->isAttached = true;
+        // Keep track of the attached pair for future validation
+        this->attachedChildModelName = this->childModelName;
+        this->attachedChildLinkName  = this->childLinkName;
+        this->PublishJointState(this->isAttached);
+        gzdbg << "Attaching entity: " << this->detachableJointEntity
+              << std::endl;
+      }
+      else
+      {
+        gzwarn << "Child Link " << this->childLinkName
+               << " could not be found.\n";
+      }
+    }
+    else if (!this->suppressChildWarning)
+    {
+      gzwarn << "Child Model " << this->childModelName
+             << " could not be found.\n";
+    }
+  }
+
+  // only allow detaching if child entity is attached
+  if (this->isAttached)
+  {
+    if (this->detachRequested && (kNullEntity != this->detachableJointEntity))
+    {
+      // Detach the models
+      gzdbg << "Removing entity: " << this->detachableJointEntity << std::endl;
+      _ecm.RequestRemoveEntity(this->detachableJointEntity);
+      this->detachableJointEntity = kNullEntity;
+      this->detachRequested = false;
+      this->isAttached = false;
+      // Publish using the last known attached pair, then clear them.
+      this->PublishJointState(this->isAttached);
+      this->attachedChildModelName.clear();
+      this->attachedChildLinkName.clear();
+    }
+  }
+}
+
+//////////////////////////////////////////////////
+bool DynamicDetachableJoint::OnServiceRequest(const gz::msgs::AttachDetachRequest &_req,
+                                              gz::msgs::AttachDetachResponse &_res)
+{
+  GZ_PROFILE("DynamicDetachableJoint::OnServiceRequest");
+
+  // Check if the request is valid
+  if (_req.child_model_name().empty() || _req.child_link_name().empty() )
+  {
+    _res.set_success(false);
+    _res.set_message("Invalid request: child_model_name and child_link_name must be set.");
+    return true;
+  }
+
+  if (_req.command().empty())
+  {
+    _res.set_success(false);
+    _res.set_message("Invalid request: command must be 'attach' or 'detach'.");
+    return true;
+  }
+
+   // If attach is requested
+   if (_req.command() == "attach")
+   {
+     if (this->isAttached) 
+     {
+       _res.set_success(false);
+      _res.set_message("Already attached to child model " + this->attachedChildModelName +
+                       " at link " + this->attachedChildLinkName + ".");
+       gzdbg << "Already attached" << std::endl;
+       return true;
+     }
+
+    // set the child model and link names from the request
+    this->childModelName = _req.child_model_name();
```

Comment:
Knit: I don't think atomics are sufficient here. You do need to use a lock to guarantee data consistency between
`this->childModelName`, `this->childModelName` and `this->attachRequested`.